### ISSUE TITLE
Issue reading file sqpack file.

### ISSUE THREAD

**Issue by akuseru** [NotAdam/Lumina#15]
Working on updating the model file readers but running into the following issue when reading a mdl file:

```
File: bg/ffxiv/sea_s1/twn/s1t1/bgplate/0032.mdl

System.IO.InvalidDataException: Found invalid data while decoding.
   at System.IO.Compression.InflaterZlib.Inflate(FlushCode flushCode)
   at System.IO.Compression.InflaterZlib.ReadInflateOutput(Byte[] outputBuffer, Int32 offset, Int32 length, FlushCode flushCode, Int32& bytesRead)
   at System.IO.Compression.InflaterZlib.Inflate(Byte[] bytes, Int32 offset, Int32 length)
   at System.IO.Compression.DeflateStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.IO.Stream.InternalCopyTo(Stream destination, Int32 bufferSize)
   at System.IO.Stream.CopyTo(Stream destination)
   at Lumina.Data.SqPackStream.ReadFileBlock(Int64 offset, MemoryStream dest, Boolean resetPosition) in \git\Lumina\src\Lumina\Data\SqPackStream.cs:line 304
   at Lumina.Data.SqPackStream.ReadFileBlock(MemoryStream dest, Boolean resetPosition) in \git\Lumina\src\Lumina\Data\SqPackStream.cs:line 282
   at Lumina.Data.SqPackStream.ReadModelFile(FileResource resource, MemoryStream ms) in \git\Lumina\src\Lumina\Data\SqPackStream.cs:line 171
   at Lumina.Data.SqPackStream.ReadFile[T](Int64 offset) in \git\Lumina\src\Lumina\Data\SqPackStream.cs:line 89
   at Lumina.Data.SqPack.ReadFile[T](Int64 offset) in \git\Lumina\src\Lumina\Data\SqPack.cs:line 123
   at Lumina.Data.Category.GetFile[T](ParsedFilePath path) in \git\Lumina\src\Lumina\Data\Category.cs:line 124
   at Lumina.Data.Repository.GetFile[T](Byte cat, ParsedFilePath path) in \git\Lumina\src\Lumina\Data\Repository.cs:line 79
   at Lumina.Data.Repository.GetFile[T](String cat, ParsedFilePath path) in \git\Lumina\src\Lumina\Data\Repository.cs:line 67
   at Lumina.Lumina.GetFile[T](String path) in \git\Lumina\src\Lumina\Lumina.cs:line 148
```

I am digging into it, but any insight into this issue would be helpful. 

Thanks :D

**Comment by NotAdam** [comment]
this is likely related to some changes that were made so files can be written and loaded by the game, as a result some of the model file loading changed in sqpackstream and mdlfile has broken as a result

**Comment by NotAdam** [comment]
@lmcintyre would know more about this than i do

**Comment by akuseru** [comment]
I think this is fixed in pull request #16. Please take a look.

**Comment by NotAdam** [comment]
looks good, thanks for that
